src/test/java/modification/ModifyTests.java

```diff
@@ -36,13 +38,25 @@ void customModifierWorks() {
     assertEquals("banana", test.foo);
   }
 
+  @Test
+  void customModifierThrows() {
+    var test = new CustomModifierThrowTest("baFOOnaFOOna");
+    assertThrows(RuntimeException.class, test::modify);
+  }
+
   @Test
   void nestedModifierWorks() {
     var test = new NestedModifierTest(new NestedModifierTest.Child("    trim    "));
     test.modify();
     assertEquals("trim", test.child.foo);
   }
 
+  @Test
+  void nestedModifierThrows() {
+    var test = new NestedModifierThrowTest(new NestedModifierThrowTest.Child("    trim    "));
+    assertThrows(RuntimeException.class, test::modify);
```

Comment:
So in these tests we're checking if we propagated thrown exception correctly?
Do we maybe want to check if the message is `DONE GOOFD` as well?

Reply:
It's a runtime exception so I don't think we would gain much asserting the message, this mostly serves as a proof of concept that tells us that if the user creates a modifier that throws, they will be faced with an exception.

Checked exceptions are also an option, but would not help much if a user indexes out of bounds in a modifier for example.